### ISSUE TITLE
[Root] DNS filtering doesn't work in Local HTTP proxy auto mode

### ISSUE THREAD

**Issue by Nickdv9** [AdguardTeam/AdguardForAndroid#2823]
Host files like StevenBlacks lists/Energized lists are not getting recognized as DNS filters and don't work when imported.

### Steps to reproduce
1. Import with the following host file in DNS Filtering -> DNS Filters.
`https://raw.githubusercontent.com/EnergizedProtection/block/master/blu/formats/hosts`

2. The following popup appears when importing the URL:
    `Downloaded filter doesn't looks like a DNS filter. Are you sure you want to continue? `
    OK to continue.

3. Test using the following methods:
    Test #1: Since energized lists blocks crashlytics, check if it blocks he following in log
                    -`https://settings.crashlytics.com`
                    -`https://e.crashlytics.com`

    Test #2: Check if the list is properly applied using their webpage [https://energized.pro/check/](https://energized.pro/check/)

4. Both the above tests fail in the current nighlty.

### Expected behavior
The above two tests should pass and the list should be imported without any issue ( I.e without the popup in step 2 )

### Actual behavior
Host files do not work when imported.

### Alternative workaround
One way of getting the same list to work would be to import the same energized list into Content Blocking -> Filters -> Custom Filters, but with the filter format ( Energized lists are available in multiple formats ).
Here's the same list as mentioned above but with a filter format: `https://raw.githubusercontent.com/EnergizedProtection/block/master/blu/formats/filter`

### Environment
* Adguard version: v3.1.35η
* Adguard filtering mode: Local HTTP Proxy
* Device model: Redmi Note 3
* Operating system and version: Android 9.0
* Root access: Yes
* Any other apps that can start VPN / Antiviruses / Cleaners etc. installed?: No

**Comment by admitrevskiy** [comment]
@Nickdv9 hi!
I've just tried to reproduce the issue and everything works fine. 
The only difference is that I use VPN-mode. Please, try to switch filtering mode to VPN and try one more time.

**Comment by ameshkov** [comment]
@admitrevskiy well, we should check why DNS filtering doesn't do anything in root mode anyways.

**Comment by Nickdv9** [comment]
@admitrevskiy you're right. It works in localVPN mode. When I switch to back to local http root mode, it stops working

**Comment by TPS** [comment]
Could 1 of y'all update the issue title to reflect that it's specific to root mode?  I had to read nearly all the way down this far to figure out why _my_ StevenBlack subscription _is_ working.…

**Comment by admitrevskiy** [comment]
@Nickdv9 I think, I've found the root cause of this issue. What kind of network (IPv4/IPv6) do you use?

**Comment by Nickdv9** [comment]
@admitrevskiy It's ipv4

**Comment by admitrevskiy** [comment]
@Nickdv9 could you please do the following: 

- Go to Settings - Advanced - Logging level -> set it to "record everything"
- Enable DNS filtering, EnergizedProtection FIlter  and choose System server
- Enable protection  
- Open `https://energized.pro/check/` in incognito mode
- Get back to DNS filtering module and select any custom DNS provider
- Wait until protection restarts
- Open `https://energized.pro/check/` in incognito mode one more time 
- Go to settings - advanced - export logs
- Send the logs to devteam@adguard.com

BTW, please, check your connection to DNS server:
- Select AdGuard DNS provider
- Go to `https://adguard.com/en/adguard-dns/overview.html`
- Scroll page down to `AdGuard DNS servers map` section and let me know if you are connected to AG DNS server

**Comment by Nickdv9** [comment]
@admitrevskiy While I was about to provide the logs and performing the said steps, I found that the issue seems to have been fixed even in root mode. I am not sure myself how it's working. The only thing that has changed since back then is that now I am on .37 version of the nightly.
Also I have been using AdGuard DNS for a while now with DNSCrypt and it is connected properly to the adguard servers.
While I am not sure yet, if the issue is completely fixed. I'd like to have this thread open for a couple more days until I am sure it's working fine, if that is okay. Thank you for all your assistance.
